virtualizarr/readers/dmrpp.py

```diff
@@ -0,0 +1,647 @@
+                if c == "shuffle":
+                    filters.append({"id": "shuffle", "elementsize": dtype.itemsize})
+                elif c == "deflate":
+                    filters.append(
+                        {
+                            "id": "zlib",
+                            "level": int(
+                                chunks_tag.attrib.get(
+                                    "deflateLevel", self._default_zlib_value
+                                )
+                            ),
+                        }
+                    )
+            return filters
+        return None
+
+    def _parse_chunks_dimensions(
+        self, chunks_tag: ET.Element
+    ) -> tuple[int, ...] | None:
+        """
+        Parse the chunk dimensions from a DMR chunks tag. Returns None if no chunk dimensions are found.
+
+        Parameters
+        ----------
+        chunks_tag : ET.Element
+            An ElementTree Element with a <chunks> tag.
+
+        Returns
+        -------
+        tuple[int, ...] | None
+
+        """
+        chunk_dim_tag = chunks_tag.find("dmr:chunkDimensionSizes", self._ns)
+        if chunk_dim_tag is not None and chunk_dim_tag.text is not None:
+            # 1 1447 2895 -> (1, 1447, 2895)
+            return tuple(map(int, chunk_dim_tag.text.split()))
+        return None
+
+    def _parse_chunks(
+        self, chunks_tag: ET.Element, chunks_shape: tuple[int, ...]
+    ) -> ChunkManifest:
+        """
+        Parse the chunk manifest from a DMR chunks tag.
+
+        Parameters
+        ----------
+        chunks_tag : ET.Element
+            An ElementTree Element with a <chunks> tag.
+
+        chunks : tuple
+            Chunk sizes for each dimension. E.g. (1, 1447, 2895)
+
+        Returns
+        -------
+        ChunkManifest
+        """
+        chunkmanifest: dict[ChunkKey, object] = {}
+        default_num: list[int] = (
+            [0 for i in range(len(chunks_shape))] if chunks_shape else [0]
+        )
+        chunk_key_template = ".".join(["{}" for i in range(len(default_num))])
+        for chunk_tag in chunks_tag.iterfind("dmr:chunk", self._ns):
+            chunk_num = default_num
+            if "chunkPositionInArray" in chunk_tag.attrib:
+                # "[0,1023,10235]" -> ["0","1023","10235"]
+                chunk_pos = chunk_tag.attrib["chunkPositionInArray"][1:-1].split(",")
+                # [0,1023,10235] // [1, 1023, 2047] -> [0,1,5]
+                chunk_num = [
+                    int(chunk_pos[i]) // chunks_shape[i]
+                    for i in range(len(chunks_shape))
+                ]
+            # [0,1,5] -> "0.1.5"
+            chunk_key = ChunkKey(chunk_key_template.format(*chunk_num))
+            chunkmanifest[chunk_key] = {
+                "path": self.data_filepath,
+                "offset": int(chunk_tag.attrib["offset"]),
+                "length": int(chunk_tag.attrib["nBytes"]),
+            }
+        return ChunkManifest(entries=chunkmanifest)
```

Comment:
Do you think you could rewrite this to use `ChunkManifest.from_arrays()`? It would potentially be a lot more performant. (But this could also be left to a later PR.)

Reply:
Oh nice didn't know about that function. I will give it a try